### PR TITLE
Create CI infrastructure in GitHub Actions

### LINKED ISSUE #335: CI is not run in PRs
**What happened**:
When a PR is opened, Circle CI is not reporting the status of the CI. I was not able to find the report in [the KUTTL Circle CI](https://app.circleci.com/pipelines/github/kudobuilder/kuttl?filter=all).

### PR DESCRIPTION
**Pull request by iblancasa**
Signed-off-by: Israel Blancas <iblancas@redhat.com>

**What this PR does / why we need it**:
* Fixes #335: CI is not run in PRs
* Use GitHub Actions for the CI
* Add a job for the unit tests (missing in the CircleCI automation)

**Some notes**
* Create as a draft because I need to add a workflow to create releases

### CODE REVIEW

**Comment by iblancasa** [comment]
@kensipe, do you want me to remove the `.circleci` folder?

**Comment by kensipe** [comment]
oh.. nice!   We may want to give a heads up and align with kudo... but would limit this to < a week.   I don't see any challenges but want to allow for alignment time.

and to answer the question.. yes!   we should remove `.circleci` as part of this transition

Nice stuff!

**Comment by iblancasa** [comment]
Ok, nice.
Another question: do you want to create a GitHub workflow for the releases or do you want to keep the process explained in `RELEASE.md`?

**Comment by kensipe** [comment]
it would be much nicer to have releases off github actions.  
some details that may not be on the release doc.   I build a docker build with buildx (multi-arch support) and push to docker hub.  This is to support Operator SDK score card.   It would be wonderful to automate all the things.

**Comment by kensipe** [comment]
I see all the details are there either on the release doc.. or linked from it.

**Comment by kensipe** [comment]
this is marked "Draft"... let me know when you're ready.  It looks good.  I expect we want to pull from circleci... is that the only thing left to change?   I would suggest that we can have additional features (like the release process) on a separate PR.  Thanks for this!

**Comment by kensipe** [comment]
@iblancasa checking to see if this is ready?

**Comment by kensipe** [comment]
@iblancasa do you think that is ready?

**Comment by kensipe** [comment]
renewed interest in this... I can take it over if needed

**Comment by eddycharly** [comment]
I didn't know about this PR, I made something super simple (only runs the linter and unit tests).
I failed to get integration tests working because TestEnv startup failure :(

**Comment by iblancasa** [comment]
Ok, this PR was originally to create the releases too. How about just fixing the PRs and create another PR for the releases?

**Comment by eddycharly** [comment]
@iblancasa your PR looks great, are we going to merge it ?
Creating release can come in another PR IMHO.
This is critical to get CI running on PRs.

**Comment by iblancasa** [comment]
@eddycharly if everyone agrees, we can merge it.
I just pushed the branch with the latest changes. Also, removed the `draft` status.

**Comment by eddycharly** [comment]
Outch all is red !

**Comment by eddycharly** [comment]
This would be great to have !

File: hack/install-kube-apiserver.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+echo Installing kube-apiserver
+
+sudo mkdir -p /usr/local/kubebuilder/bin
+
+sudo curl -L https://dl.k8s.io/v1.16.4/kubernetes-server-linux-amd64.tar.gz -o /usr/local/kubebuilder/bin/kubernetes-server-linux-amd64.tar.gz
```
Comment (eddycharly) [inline hack/install-kube-apiserver.sh:6]: This is k8s 1.16 ?

Comment (erikgb) [reply hack/install-kube-apiserver.sh:6]: I'll guess so. I suggest installing etcd and apiserver using https://github.com/kubernetes-sigs/controller-runtime/tree/master/tools/setup-envtest in a follow-up MR - that can also upgrade to a supported K8s version.

Comment (iblancasa) [reply hack/install-kube-apiserver.sh:6]: I'm not sure. I ported what the previous automation had. I think we can do the upgrade later as @erikgb suggested

**Comment by kensipe** [comment]
very interested in this PR and those related.   I will be reviewing today and pushing to merge asap

**Review by erikgb — COMMENTED** [review]
In general LGTM, but I would prefer if the setup task were better integration into the Makefile, to allow it to be run on local developer workstation. But that can be improved in a follow-up PR. As mentioned in a comment, we also need to upgrade the K8s version. K8s 1.16 is so totally unsupported.

I also think we should remove the `.circleci`, as having multiple CI configs will appear confusing.

**Comment by kensipe** [comment]
agree with @erikgb   however I'm good with landing this and removing circle in a separate PR.   reviewing... however we will likely want to review the actual experience.   Reviewing content now.

File: hack/install-kube-apiserver.sh (hunk 1 above)
Comment (kensipe) [reply hack/install-kube-apiserver.sh:6]: agree with the team that we should upgrade the k8s version... but aligned with iblancasa that we can do that on the next PR

**Review by kensipe — APPROVED** [review]
/lgtm.  thank you so much @iblancasa !    I will be merging today.   Based on tests, I will create a PR to remove circle.   Following that we can look to bump versions to something more modern.

**Comment by kensipe** [comment]
@eddycharly  we use api-server and etc for integration tests... we use `kind` for e2e tests... it seems we are covered there

**Comment by kensipe** [comment]
Actions to follow:
https://github.com/kudobuilder/kuttl/issues/427
https://github.com/kudobuilder/kuttl/issues/428
https://github.com/kudobuilder/kuttl/issues/429

**Comment by eddycharly** [comment]
> we use kind for e2e tests

where does this happen ?

**Comment by eddycharly** [comment]
In the e2e logs:
```
harness.go:220: started test environment (kube-apiserver and etcd) in 4.821556848s
```

**Pull request merged**